pkg/storagecapabilities/storagecapabilities.go

```diff
@@ -269,17 +269,17 @@ var storageClassToProvisionerKeyMapper = map[string]func(sc *storagev1.StorageCl
 		return "pxd.openstorage.org"
 	},
 	"kubernetes.io/portworx-volume": func(sc *storagev1.StorageClass) string {
-		val := sc.Parameters["shared"]
-		if val == "true" {
-			return "kubernetes.io/portworx-volume/shared"
+		val := sc.Parameters["sharedv4_mount_options"]
+		if strings.Contains(val, "vers=3.0") && strings.Contains(val, "nolock") {
+			return "kubernetes.io/portworx-volume/nfs"
 		}
 		return "kubernetes.io/portworx-volume"
 	},
 	"pxd.portworx.com": func(sc *storagev1.StorageClass) string {
-		// https://docs.portworx.com/portworx-install-with-kubernetes/storage-operations/csi/volumelifecycle/#create-shared-csi-enabled-volumes
-		val := sc.Parameters["shared"]
-		if val == "true" {
-			return "pxd.portworx.com/shared"
+		// https://docs.portworx.com/portworx-enterprise/operations/operate-kubernetes/storage-operations/manage-kubevirt-vms.html#create-a-storageclass
+		val := sc.Parameters["sharedv4_mount_options"]
+		if strings.Contains(val, "vers=3.0") && strings.Contains(val, "nolock") {
```

Comment:
Maybe splitting over , and asserting that those exist will be more robust/clean?

Reply:
Afaiu these `nolock/3.0` params are mostly for backward compatibility with older versions supporting `RWX Filesystem` only, and soon all releases will prefer `RWX Block`, so adding `StorageClass` params seems odd. @adityadani @gohilankit wdyt?

Reply:
I mean an indicator for RWX/Block

Reply:
I guess it will be preferred by all?

Reply:
We don't have an indicator for `RWX Block` in StorageClass. It's based on the accessModes & volumeMode on PVC. So just the absence of above mentioned params in StorageClass can indicate it's `RWX Block`.

Reply:
I meant that we would add one